src/app/clusters/door-lock-server/door-lock-server.h

```diff
@@ -188,6 +189,17 @@ class DoorLockServer
 
     inline bool SupportsUnbolt(chip::EndpointId endpointId) { return GetFeatures(endpointId).Has(Feature::kUnbolt); }
 
+    /**
+     * @brief Allows the application to register a custom callback which will be called before the default DoorLock
```

Comment:
I assume this "before the default" bit is why the application can't just register with the fabric table directly?

Reply:
The default one only invalidates the credentials, my goal was to extend the possibilities here. Registering this callback in the door lock sever directly seems reasonable to me.

We may also run the callback after the default code, maybe this would allow the application to detect if additional action is in fact needed.

Reply:
@bzbarsky-apple does this resolve your comment?

Reply:
> Registering this callback in the door lock sever directly seems reasonable to me.

If we don't care about ordering wrt door lock server, registering directly with the fabric table seems fine too...  But I am fine with this setup too.  It's better to do "after", since then it guarantees the callee will not blow away state the door lock server itself may need for cleanup.

Reply:
But I guess I am wondering why we need the extra code, if we don't care about the ordering, instead if just having the cases that need it register directly on the fabric table @markaj-nordic

Reply:
I mean, we actually do care about the order because of what you pointed out - it's better for the callee to be able to check the state of the credentials based on received fabric idx and then act accordingly. Also, moving the callback "after" makes us sure that our changes to the door-lock internal state in the application will not be overwritten later (when removing fabric).